ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0104__rhods_operator/0114__component_images.robot

```diff
@@ -0,0 +1,106 @@
+*** Settings ***
+Documentation       Test Suite to check for components
+
+Resource            ../../../../../../tasks/Resources/RHODS_OLM/install/oc_install.robot
+
+Suite Setup         Suite Setup
+Suite Teardown      Suite Teardown
+
+
+*** Variables ***
+@{COMPONENTS} =
+...                 codeflare
+...                 modelregistry
+...                 trustyai
+...                 ray
+...                 kueue
+...                 workbenches
+...                 dashboard
+...                 modelmeshserving
+...                 datasciencepipelines
+...                 trainingoperator
+
+
+*** Test Cases ***
+Check For Correct Component Images
+    [Documentation]    The purpose is to enforce the check of correct component images on their deployments.
+    [Tags]                  Operator                RHOAIENG-12576
+    FOR    ${cmp}    IN    @{COMPONENTS}
+        ${rc}    ${managementState}=    Run And Return Rc And Output
+        ...    oc get DataScienceCluster default-dsc -o jsonpath="{.spec.components['${cmp}'].managementState}"
+        IF    "${managementState}" == "Managed"
+            IF    "${cmp}" == "codeflare"
+                Check Image On Csv And Deployment
+                ...    odh_codeflare_operator_image
+                ...    codeflare-operator-manager
+            ELSE IF    "${cmp}" == "modelregistry"
+                Check Image On Csv And Deployment
+                ...    odh_mlmd_grpc_server_image
+                ...    model-registry-operator-controller-manager
+                Check Image On Csv And Deployment
+                ...    odh_model_registry_image
+                ...    model-registry-operator-controller-manager
+            ELSE IF    "${cmp}" == "trustyai"
+                Check Image On Csv And Deployment
+                ...    odh_trustyai_service_operator_image
+                ...    trustyai-service-operator-controller-manager
+            ELSE IF    "${cmp}" == "ray"
+                Check Image On Csv And Deployment
+                ...    odh_kuberay_operator_controller_image
+                ...    kuberay-operator
+            ELSE IF    "${cmp}" == "kueue"
+                Check Image On Csv And Deployment
+                ...    odh_kueue_controller_image
+                ...    kueue-controller-manager
+            ELSE IF    "${cmp}" == "workbenches"
+                Check Image On Csv And Deployment
+                ...    odh_notebook_controller_image
+                ...    odh-notebook-controller-manager
+                Check Image On Csv And Deployment
```

Comment:
cant we do on a different way to avoid a statement per component?

Reply:
No, robot framework has limitations on this. I was investigating

Reply:
we can try to have a variable to map the component name to the image name, like this:

https://github.com/red-hat-data-services/ods-ci/blob/10015041eb960938d5fae2c6673787ae2b558728/ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0104__rhods_operator/0113__dsc_components.robot#L41

Reply:
you could try using the [Template] approach. Check `Verify Pipelines Integration With Elyra When Using Standard Data Science Based Images` test case for an example or the official RF user guide